### PR TITLE
dev-libs/keystone: use get_libdir for multilib

### PR DESCRIPTION
**Pull request by sielicki**
Package-Manager: Portage-2.3.64, Repoman-2.3.12
Signed-off-by: Nicholas Sielicki <sielicki@yandex.com>

### CODE REVIEW

**Comment by sielicki** [comment]
fixes: https://bugs.gentoo.org/652440

**Comment by gyakovlev** [comment]
keystone is simply not installable as ebuild is incorrect, not sure why it even got into the tree.

it does not specify llvm dep or llvm max slot. not your fault of course, don't take it personally.

try installing it on a system without llvm and it will blow up

```shell-session
 * ERROR: dev-libs/keystone-0.9.1::gentoo failed (setup phase):
 *   No LLVM slot found installed!
 * 
 * Call stack:
 *     ebuild.sh, line 124:  Called pkg_setup
 *     ebuild.sh, line 369:  Called llvm_pkg_setup
 *   llvm.eclass, line 163:  Called get_llvm_prefix ''
 *   llvm.eclass, line 142:  Called die
 * The specific snippet of code:
 *      die "No LLVM slot${1:+ <= ${1}} found installed!"

```



also it should have profile entries to have llvm_targets on relevant arches by default (like rust or llvm itself has)
```
arch/amd64/package.use.force
6:dev-lang/rust llvm_targets_X86
23:dev-ml/llvm-ocaml llvm_targets_X86
```

can you also check those issues?

**Comment by gyakovlev** [comment]
filed a very cool bug number for llvm issue =)
https://bugs.gentoo.org/685000

**Comment by gyakovlev** [comment]
ping

**Comment by mgorny** [comment]
> it does not specify llvm dep or llvm max slot. not your fault of course, don't take it personally.

Lack of max slot is valid if -9999 is acceptable. However, missing dep is a problem.

**Comment by juippis** [comment]
@sielicki @nemunaire whats the status here?

**Comment by sielicki** [comment]
Apologies for disappearing for a few *months*... my dog ate my homework.

@gyakovlev 

> it does not specify llvm dep or llvm max slot

According to https://wiki.gentoo.org/wiki/Project:LLVM/Ebuild_Guide you don't have to specify max slot if 9999 is supported:

> LLVM_MAX_SLOT needs to be set to the highest supported LLVM slot (or unset if there is no limit, i.e. 9999 is supported).

Just pushed a commit that fixes the lack of a dependency in the non-live ebuild and adds keystone to the profile `package.use`s.

Sorry again about leaving like this.

**Comment by gyakovlev** [comment]
pong, I was away. will look at it closer within couple of days.

**Comment by linxon** [comment]
bump

**Comment by linxon** [comment]
@gyakovlev Hi! I see the low activity of this

Maybe I need to rework this with new PR?

**Comment by sielicki** [comment]
@linxon I never received feedback, patch was fine AFAIK.

File: dev-libs/keystone/keystone-0.9.1.ebuild [hunk 1]
```diff
@@ -5,7 +5,7 @@ EAPI=6
 
 DISTUTILS_OPTIONAL=1
 CMAKE_MIN_VERSION=3.8.7
-PYTHON_COMPAT=( python{2_7,3_5,3_6} )
+PYTHON_COMPAT=( python{2_7,3_5,3_6,3_7} )
```
Comment (linxon) [inline dev-libs/keystone/keystone-0.9.1.ebuild]: I guess it needs to be updated

**Comment by sielicki** [comment]
I really wanted to get around to doing this over the weekend but I'm too busy with work at the moment to work on this. Feel free to take this over.

**Comment by sielicki** [comment]
I've just rebased this onto master, added some bumps in python version.

Note that I'm flying blind here (don't have a gentoo install at the moment) so test/review is definitely needed. I'd really like to see this PR/bug closed, though.

**Comment by sielicki** [comment]
Reviewing CI:

**Comment by gyakovlev** [comment]
actually it's good that you closed/reopened it, it completely lost it, it should have been reviewed/fixed long ago. for some reason github just did not send me anything till you closed it. looking.

mind if I just commit fixes based on this PR? I don't want to go back and forth again till it fixed and will just do the work myself.

**Comment by sielicki** [comment]
Sure, that'd be totally fine with me, just happy to see this get taken care of finally! Thanks so much.

**Comment by sielicki** [comment]
@gyakovlev I'll also note that looking at the [keystone commits](https://github.com/keystone-engine/keystone/commits/master), he's really close to releasing 0.9.2, so if you wanted to put it off a few more days it might be a good idea to consolidate effort.

**Pull request closed without merging**